ships/0033-build-output-vulnerability-scanning.md

```diff
@@ -0,0 +1,238 @@
+
+If the `fail` flag is set to `true` and a vulnerability is found that is not ignored, then the image-processing step will exit with status code 22. This is an arbitrary choice - or the 22nd letter of the alphabet is v, for vulnerable. The push operation is then not happening.
+
+### Configuration
+
+One nice feature of Trivy is that options can be provided using environment variables as an alternative to command line flags. This will enable Shipwright administrators to enable features like the Redis-based caching backend (`TRIVY_CACHE_BACKEND`), a custom location of the remote vulnerability database (`TRIVY_DB_REPOSITORY`), or an air-gapped mode (`TRIVY_SKIP_DB_UPDATE` and `TRIVY_OFFLINE_SCAN`) in the Shipwright configuration in the container template of the image-processing step. We can document this but will state that no API-level guarantee is given here because Trivy might change, or Shipwright might go away from Trivy.
+
+### CLI
+
+The new output settings will be added as flags to the appropriate CLI commands.
+
+### Sample updates
+
+None
+
+### Documentation updates
+
+The Build and BuildRun documentation are updates to explain the new fields and their behavior.
+
+### Test Plan
+
+The implementation has to be tested on a `unit`, `integration` and `e2e` level to ensure correctness.
+
+## Release Criteria
+
+## Risks and Mitigations
+
+- Risk: Trivy might become unmaintained and outdated. Mitigation: Trivy is nowhere exposed to users. The data that we will present in the BuildRun status and in the container logs is independent of the tool. A switch to a different tool (see also [Alternatives](#alternatives)) should be possible without any problems.
+
+## Drawbacks
+
+## Alternatives
+
+### Unfixed vulnerabilities
+
+Instead of providing a flag in the Build to ignore unfixed vulnerabilities, we could also consider ignoring them always. Based on my observation, security scanning in container registries do that. Just run for example `trivy image registry.access.redhat.com/ubi8/ubi-minimal`, it lists currently 67 unfixed vulnerabilities, two of them high. On the other hand, `trivy image alpine` returns no vulnerabilities. `trivy image ubuntu:focal` finds 18 vulnerabilities but none of them high.
+
+### Using a different tool
+
+As an alternative to Trivy, the following tools could be considered:
+
+- [Grype](https://github.com/anchore/grype), is not considered at the moment because it is slower (see [comparison](#performance-comparison-between-trivy-and-grype)) and I think finds less because it does not scan as much language-specific things (Go modules, node modules, etc). But in general, it is also actively maintained and improved with high velocity.
+- [Clair](https://github.com/quay/clair), is not considered because its architecture is distributed and cannot run standalone.
```

Comment:
@crozzy may want to chime in here.

Reply:
Thanks @ricardomaraschini! @SaschaSchwarze0 we are working on a standalone Clair cli for use in the CI. Currently it is being molded into a github action but it feels like it might slot in here well too.

Reply:
That's fair. I think I mentioned somewhere that the design is allowing a change of the tool at any point (or even running more than one tool if we think that is valuable). Based on that, if we ever consider a tool being better than the current choice, then we can always revisit it.